### DISCUSSION TITLE
The frame rate is particularly low when calling HandleTopologyChange()

### DISCUSSION THREAD

**Discussion by RickYang1** [InfinyTech3D/SofaUnity#115] [Help - Using SofaUnity] [answered]
Hello,
I need to use the tetrahedral removal function of `TopologicChangeProcessor` multiple times in my simulation. After performing tetrahedral removal once, I need to call `HandleTopologyChange()` to update the tetrahedral information of the model. However, I found that frequently calling this function increases the computation time of `sofaPhysicsAPI_step()` and reduces the frame rate:
![image](https://github.com/user-attachments/assets/070aec72-bad0-401b-9b78-c6b1eeeb5a8d)
How can I solve it?
Thanks in advance.

**Comment by epernod** [comment, marked as answer]
Hi,

this is normal because this method is recomputing the whole Unity mesh from new SOFA mesh information. I.e getting all the new mesh structure (points, triangles and tetrahedra) and recomputing all the map to create the Unity MeshFilter.

Right now there is nothing you can do. The idea would be to improve those methods, maybe by using multi threaded communication and mesh copy. This is unfortunately not on our roadmap for the moment.
